src/main/java/org/folio/services/transactions/BatchTransactionService.java

```diff
@@ -70,4 +80,35 @@ private Future<Void> updateTransaction(Transaction transaction, RequestContext r
       .withTransactionsToUpdate(singletonList(transaction));
     return processBatch(batch, requestContext);
   }
+
+  private Future<Void> checkDeletions(Batch batch, RequestContext requestContext) {
+    if (batch.getIdsOfTransactionsToDelete().isEmpty()) {
+      return succeededFuture();
+    }
+    return anyConnectedToInvoice(batch.getIdsOfTransactionsToDelete(), requestContext)
+      .map(connected -> {
+        if (FALSE.equals(connected)) {
```

Comment:
**Question**
Does sonar generates warning for `!connected` syntax? I think `connected` won't be null here.

Reply:
Yes it does. I had to correct "!connected" to avoid that warning (but I agree with you, it won't be null).